### PR TITLE
feat(left-nav): refactor into left-nav-sections and include animations

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Web Component: Change Mega Menu to include animation (back button & forward motion) #4831
[Left Nav: Web Components]: L1 items are visible when on L2 menu items #6209
Web component: Masthead-With custom data - Hybrid cloud options are missing from the menu options. #5135

### Description

Refactor `left-nav` so all menu panels/sections are on the same level instead of nested. 

Align animations for forward and backward movement.

Jun-16-2021 17-42-51

### Changelog

**New**

- `left-nav-menu-section` component that handles all the focus of first element, back button functionality
- add L1 platform name to the left-nav

**Changed**

- set tabindex to -1 for `left-nav-menu` and `left-nav-menu-item`
- reduced logic of `left-nav-menu` to only render menu items that have a menu
- move cloud profile and cta menu items to within a menu section in `cloud-masthead-composite`
- add conditional to `focuswrap` utility when focus doesn't hit start or end sentinel which allows the first menu item to focus when left-nav is expanded

**Removed**

- styles that are no longer needed

### CODE REVIEW

**Comment by shixiedesign** [comment]
Thank you so much for taking this one! 

1. Focus state of `Back` button should keep the blue color. Currently it becomes black on focus
Jun-24-2021 13-10-56

2. When I `tab` through all the options in the panel, the focus state disappears and next panel appears, however I cannot get the focus state back, and `left` and `right` arrow keys slide the panels back and forth. I imagine the disappearing focus state is important to fix. In react, the focus state returns to menu's top item. 
Jun-24-2021 14-23-56

3. In both web comp & react: how do I navigate to `x` button with keyboard? `Tab` never gets there it seems. 

4. In both web comp & react, `md` breakpoint: clicking outside the menu on page with an overlay should close the menu. Currently it doesn't do anything. 

5. At the bottom of the menu, there should always be 48px spacing. Currently missing bottom spacing at the end of the menu:  
Jun-24-2021 14-33-15

6. Panels allow me the scroll left-and-right: 
Jun-24-2021 14-30-49

**Comment by annawen1** [comment]
> Thank you so much for taking this one!
> 
> 1. Focus state of `Back` button should keep the blue color. Currently it becomes black on focus

Done

> 2. When I `tab` through all the options in the panel, the focus state disappears and next panel appears, however I cannot get the focus state back, and `left` and `right` arrow keys slide the panels back and forth. I imagine the disappearing focus state is important to fix. In react, the focus state returns to menu's top item.

It should now focus on the back button when on the next panel. Also the `left` and `right` arrow keys should not allow for the panels to slide back and forth anymore. 

> 3. In both web comp & react: how do I navigate to `x` button with keyboard? `Tab` never gets there it seems.

In web components, you should be able to get to the `x` button when tabbing through the menu items now, it should jump to the button when reaching the last menu item. 

For React, it's coded a bit differently and user has to go back to the initial menu panel to access the `x`.

> 4. In both web comp & react, `md` breakpoint: clicking outside the menu on page with an overlay should close the menu. Currently it doesn't do anything.

hmm yeah that's been like that since we created the left nav I believe. We can address this in another ticket.

> 5. At the bottom of the menu, there should always be 48px spacing. Currently missing bottom spacing at the end of the menu:

Done

> 6. Panels allow me the scroll left-and-right:

Addressed!

**Comment by shixiedesign** [comment]
1. `shift+Tab` allows users to access the Profile and Search buttons. Expecting keyboard access to not reach there just like mouse is unable to interact with those: 
Jun-25-2021 13-20-46

2. Both react & web components: the background overlay should fade out over time with the menu moving out. Currently the background overlay just disappears. 
Jun-25-2021 13-22-32

File: packages/web-components/src/components/masthead/left-nav-menu-section.ts [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+  /**
+   * A selector selecting tabbable nodes.
+   */
+  static get selectorTabbable() {
+    return [
+      selectorTabbable,
+      'dds-left-nav-item:not([tabindex="-1"])',
+      'dds-left-nav-menu:not([tabindex="-1"])',
+      'dds-left-nav-menu-item:not([tabindex="-1"])',
```
Comment (kennylam) [inline packages/web-components/src/components/masthead/left-nav-menu-section.ts:258]: Just to stay consistent with previous interpolation.

```suggestion
      `${ddsPrefix}-left-nav-item:not([tabindex="-1"])`,
      `${ddsPrefix}-left-nav-menu:not([tabindex="-1"])`,
      `${ddsPrefix}-left-nav-menu-item:not([tabindex="-1"])`,
```

File: packages/web-components/src/components/masthead/left-nav-menu-section.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+  render() {
+    const { backButtonText, title, _handleClickBack: handleClickBack, showBackBtn } = this;
+    return html`
+      <ul>
+        ${showBackBtn
+          ? html`
+              <li class="bx--side-nav__menu-item bx--masthead__side-nav--submemu-back" role="none">
+                <button class="bx--side-nav__link" tabindex="-1" role="menuitem" @click="${handleClickBack}">
+                  <span class="bx--side-nav__link-text">${ChevronLeft20()}${backButtonText}</span>
```
Comment (kennylam) [inline packages/web-components/src/components/masthead/left-nav-menu-section.ts:214]: Just to stay consistent with previous interpolation.

```suggestion
              <li class="${prefix}--side-nav__menu-item ${prefix}--masthead__side-nav--submemu-back" role="none">
                <button class="${prefix}--side-nav__link" tabindex="-1" role="menuitem" @click="${handleClickBack}">
                  <span class="${prefix}--side-nav__link-text">${ChevronLeft20()}${backButtonText}</span>
```

**Review by kennylam — COMMENTED** [review]
Nice one @annawen1! Just some really small fixes.

**Comment by annawen1** [comment]
@shixiedesign The first issue is now addressed. The second one we can open a new ticket for.

> 1. `shift+Tab` allows users to access the Profile and Search buttons. Expecting keyboard access to not reach there just like mouse is unable to interact with those:
>    Jun-25-2021 13-20-46
> 2. Both react & web components: the background overlay should fade out over time with the menu moving out. Currently the background overlay just disappears.
>    Jun-25-2021 13-22-32

**Review by jeffchew — APPROVED** [review]
LGTM!

**Review by shixiedesign — APPROVED** [review]
🌶️ 🌶️ 🌶️

**Pull request merged**